docs/proposals/version-documentation.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "Documentation Versioning"
+linkTitle: "Documentation Versioning"
+weight: 1
+slug: documentation-versioning
+---
+
+- Author: [Jay Batra](https://github.com/jaybatra26)
+- Date: March 2020
+- Status: proposal
+
+## Problem
+
+In Cortex, currently, we are missing versioning of documentation. The idea is to have version documentation just like Prometheus.[`Prometheus`](https://prometheus.io/docs/introduction/overview/). Documentation is the main source of information for current contributors and first-timers. A properly versioned documentation will help everyone to have a proper place to look for answers before flagging it in the community.
+
+In this proposal, we want to solve this. In particular, we want to:
+
+1. Version specific pages of the documentation
+2. Include links to change version (the version must be in the URL)
+3. Include the master version and last 3 minor releases. Documentation defaults to the last minor release.
```

Comment:
Why not keep all released minor versions? I have occasionally found it useful to be able to find *very* old (10-15 years) documentation for some projects, but if cost (in terms of maintainance, work, ...) is prohibitive, then let's not do that.

Reply:
Good feedback. Given the fast pace of Cortex and that we just cut the first stable release, I was wondering if it's healthy to keep the doc for very old versions, considering we want people to use recent versions. But I don't feel strong about this, and at the end the number of versions to keep will just be a configuration option in the website generation.